core/backend/src/CheckpointManager.ts

```diff
@@ -112,6 +112,8 @@ export class Downloads {
  * @internal
 */
 export class V2CheckpointManager {
+  private static reattachDueTimestamps: { [key: string]: number } = {};

```

Comment:
Well, first of all I didn't want to add a new public property that doesn't apply to most checkpoints. If I were to add it there, I'd rather make a new sub-class for V2 checkpoints and use type checking instead of `isV2Checkpoint`. But even then, I don't have a `SnapshotDb` object in `V2CheckpointManager.attach`, I only have the props. To make it worse, when opening a V2 checkpoint, `attach` has to be called before the `SnapshotDb` is constructed, so I wouldn't be able to use `findByKey`.

Reply:
Please remove all of the findOrOpen stuff from IModelDb - that's not appropriate for anything other than RpcRead cases (and even then, i don't understand when it would ever come into play). 

Just add a `reattachDaemon` method on IModelDb that does nothing. In SnapshotDb add the timeout member when it is a V2 checkpoint and override `reattachDaemon` to update the daemon if the timeout has expired, otherwise do nothing. In `IModelReadRpcImp` add a method that finds the iModelDb and then always calls `reattachDaemon`. 

BTW, @swbsi I don't understand the `ClientReqeustContext.current` voodoo in RPC. Where does that get set?

Reply:
actually, i see `SnapshotDb.reattachDaemon` already exists. Just update it to check the timeout and add `reattachDaemon` to IModelDb so you can call it without having to do an instanceof test. 

Reattaching daemon may become relevant for `BriefcaseDb` for web editing backends sometime in the future (although they won't use the CheckpointManger to get the SAS token).

Reply:
> BTW, @swbsi I don't understand the `ClientReqeustContext.current` voodoo in RPC. Where does that get set?

It's set in RpcInvocation. The primary purpose of the request context is to propgate the request's activity id through the awaits that might occur in an RPC implementation function.

https://github.com/imodeljs/imodeljs/blob/2f92222e17eb2cd7ab0e8b09447b06a12de6ade9/core/common/src/rpc/core/RpcInvocation.ts#L123

https://github.com/imodeljs/imodeljs/blob/master/docs/learning/backend/ManagingClientRequestContext.md

Reply:
>  The primary purpose of the request context is to propgate the request's activity id 

in this case we're using it to get the AccessToken for the user from the frontend. I assume this is the right way?

Reply:
@paulius-valiunas this line (`reattachDueTimestamps`) should still be removed, right?

Reply:
should I mark this as resolved? I assume this is the right way to get the access token since that's how other RPC implementation methods do it.